### ISSUE TITLE
Sesame Does not refresh itself after new app installed or uninstalled

### ISSUE THREAD

**Issue by as-if-iam** [BranchMetrics/sesame_issues#189]
It's really tedious work to refresh manually all the time.
If I install or uninstall anything, the shortcuts are not refreshes automatically.
You can check the broadcast for Package uninstall and install to refresh the icons in search of sesame.

**Comment by philwall2** [comment]
Hey Asif, can you link me to that broadcast option? I'm pretty sure the ability for non-launcher 3rd party apps to receive package changes from the broadcast was removed in Android 8 (for security reasons I think). 

That said, if you're using Lawnchair or Hyperion launcher and have the data integration turned ON, we'll get the update from the launcher.

Gonna close this for now, but will re-open it if you have new info (that it wasn't removed in Android 8).

EDIT: here's what I was talking about https://developer.android.com/about/versions/oreo/background#broadcasts

**Comment by as-if-iam** [comment]
Thanks for the reply,

In layman's language:
So when an application is installed, the icon appears on the homepage/launcher page of nova launcher.

But when I swipe down (gesture for sesame search), the search page appears. There if I search for an application, the package isn't available. 

If there's onResume there, you can compare the current and last packaged list and see if there was anything added/removed. This is not an optimal solution.

I'll for sure get back with a better solution.

Thanks.

**Comment by philwall2** [comment]
That's a smart work around. Steve would know better than me but really the launcher should give us the data. That way we don't have to run a check every time a user swipes open their app drawer. There's a supported method for doing this that we're ready to receive from the launcher if they implement it.
